data/espers.py

```diff
@@ -159,6 +159,14 @@ def remove_all_life(self):
             if esper.has_spell(life3_id):
                 esper.remove_spell(life3_id)
 
+    def remove_top_spells(self):
+        from constants.spells import top_spells
+        for top_spell in top_spells:
+            id = self.spells.get_id(top_spell)
+            for esper in self.espers:
+                if esper.has_spell(id):
+                    esper.remove_spell(id)
```

Comment:
Instead of removing the spell here, perhaps we should replace it instead
```
if esper.has_spell(id):
  replacement = self.spells.get_random_spell(exclude = top_spells)
  esper.remove_spell(id);
  esper.add_spell(replacement)
```

Reply:
if you had each esper learning 1 spell each, it'd be awkward to find any with 0 spells available

Reply:
Good call, I think this logic could also apply to remove_all_life and remove_all_ultima, which can lead to these same sorts of gaps. I think I can refactor espers.py a little to make this type of thing work for all of those examples.